plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -194,22 +190,20 @@ private CollectedStatistics mergeStatisticsIfNecessary(
                 updateMode = REPLACE;
             }
         }
+        Map<Integer, CompactSketch> collectedNdvSketches = collectedStatistics.ndvSketches();
+        Set<Integer> columnsWithRecentlyComputedStats = new HashSet<>(collectedNdvSketches.keySet());
```

Comment:
is needed only in one case, so should inside switch below

i pushed a fixup with this (and some small other changes)